src/Umbraco.Web.UI.Client/src/views/propertyeditors/grid/editors/rte.controller.js

```diff
@@ -11,9 +11,11 @@
         vm.openEmbed = openEmbed;
 
         function openLinkPicker(editor, currentTarget, anchorElement) {
+						
             vm.linkPickerOverlay = {
                 view: "linkpicker",
                 currentTarget: currentTarget,
+				anchors: tinyMceService.getAnchorNames(JSON.stringify(editorState.current.properties)),
```

Comment:
@nathanwoulfe is there are reason editorState is being used here? There's also no null check in the getAnchorNames which causes some issues if this is undefined or null (not sure why it is but it is in the branch i'm working on). 

Surely we don't want to use editorState here, don't we want to inspect the current content of the current tinymce editor?

Reply:
@Shazwazza, yup, that's intentional - I'm passing the full property set to that service method, then pulling out any named anchors from any properties to build a dropdown list of available anchors on the current page.  Same method is used after selecting a node in the link picker.

Means I can grab anchors in other editors (grid with multiple RTEs, for example). Just going to run it through some jsperf testing to see if is unreasonably slow. Should really just be sending the property values rather than the entire property too...

Reply:
@nathanwoulfe ah ok thanks for the clarification. I've had to add a fix in my v8 branch to check for a null value... but I'm pretty sure that's because the whole content structure is different in v8 with variants so the property collection doesn't exist like that: https://github.com/umbraco/Umbraco-CMS/commit/5fdaa451b82c9701686885261cd63e628c7123a6

Reply:
@Shazwazza no worries - did some quick performance checks, it's fractionally faster to just stringify the entire property set than it is to map the values only and then stringify that array. Talking fractions of a millisecond for a large blob of content, so it's trivial really.